### PR TITLE
Add new Build site job to testing workflow and fix building site

### PR DESCRIPTION
**Pull request by raulcd**
Added new job on testing workflow to validate the site is built successfully.

Fix dialect.options rendering.

### CODE REVIEW

**Comment by raulcd** [comment]
@richtia @amol- I did some local debugging and this is the first time I am looking at this repo but the `dialect.options` is a list of options not a dict.
I've also added on this PR a job so the site is built on PRs so we don't fail rendering the site in the future.

**Comment by raulcd** [comment]
The build site job is successful as seen on the PR jobs now

File: bft/templates/scalar_function.j2 [hunk 1]
```diff
@@ -72,8 +72,7 @@
             </select>
             {% for dialect in dialects %}
             <dl class="dialect-definition" id="dialect-{{ dialect.name }}">
-                {% for opt, val in dialect.options %}
-                <dt>{{ opt }}</dt>
```
Comment (amol-) [inline bft/templates/scalar_function.j2:76]: Do we know what the output of the new changed code looks like? 
I think in the past this rendered as
```
overflow
    SILENT
rounding
    TIE_TO_EVEN
```

but if we remove the one of the two, won't we end up losing half of the information?

@sanjibansg can you confirm this?

Comment (raulcd) [reply bft/templates/scalar_function.j2:76]: Ok I see where was that, thanks for the pointer @amol- . It seems there's something that has to be updated still. If @sanjibansg can't investigate I can't try to understand where was this updated and why:
image

Comment (raulcd) [reply bft/templates/scalar_function.j2:76]: Ok, I've found this was changed on this PR: https://github.com/voltrondata/bft/pull/8/files#diff-cf88663aeb2bec1d14d86e6e1e443fea4a19e61c72cd146df36ee7fba999c206
Now it renders correctly, see from my local build:
image

**Comment by richtia** [comment]
@raulcd Thanks for fixing this!

**Pull request merged**